custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -780,25 +784,80 @@ def _prepare_prediction_server_or_batch_pipeline(self, run_language):
 
         return DrumUtils.render_file(functional_pipeline_filepath, replace_data)
 
-    def _run_predictions_pipelines_in_mlpiper(self):
-        run_language = self._check_artifacts_and_get_run_language()
+    def _run_predictions(self):
+        if self.run_mode != RunMode.SCORE:
+            raise NotImplemented("Only score mode is supported here")
 
-        if self.run_mode == RunMode.SERVER:
-            # in prediction server mode infra pipeline == prediction server runner pipeline
-            infra_pipeline_str = self._prepare_prediction_server_or_batch_pipeline(run_language)
-        elif self.run_mode == RunMode.SCORE:
-            tmp_output_filename = None
-            # if output is not provided, output into tmp file and print
-            if not self.options.output:
-                # keep object reference so it will be destroyed only in the end of the process
-                __tmp_output_file = tempfile.NamedTemporaryFile(mode="w")
-                self.options.output = tmp_output_filename = __tmp_output_file.name
-            # in batch prediction mode infra pipeline == predictor pipeline
+        with self._setup_output_if_not_exists():
+            run_language = self._check_artifacts_and_get_run_language()
             infra_pipeline_str = self._prepare_prediction_server_or_batch_pipeline(run_language)
+
+            pipeline = json.loads(infra_pipeline_str)
+            if "pipe" not in pipeline or not pipeline["pipe"]:
+                raise DrumCommonException("Pipeline is empty")
+            if "arguments" not in pipeline["pipe"][0]:
+                raise DrumCommonException("Arguments are missing in the pipeline")
+
+            self.logger.info(
+                f">>> Start {ArgumentsOptions.MAIN_COMMAND} in the {self.run_mode.value} mode"
+            )
+
+            sc = StatsCollector(
+                disable_instance=(
+                    not hasattr(self.options, "show_perf")
+                    or not self.options.show_perf
+                    or self.run_mode == RunMode.SERVER
+                )
+            )
+            sc.register_report("Full time", "end", StatsOperation.SUB, "start")
+            sc.register_report(
+                "Init time (incl model loading)", "init", StatsOperation.SUB, "start"
+            )
+            sc.register_report("Run time (incl reading CSV)", "run", StatsOperation.SUB, "init")
+            sc.enable()
+
+            try:
+                sc.mark("start")
+                predictor = GenericPredictorComponent()
+                predictor.configure(pipeline["pipe"][0]["arguments"])
```

Comment:
nit: if caller does not invoke the `configure`, the `GenericPredictorComponent` will not be properly initialized. I.e. call to `self._predictor` evaluates to NPE.